packages/components/src/shared/withListeners.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+import React from 'react';
+
+export default WrappedComponent => {
```

Comment:
I disagree with the decision to turn this into a HOC. There are no silver bullets (as the text you refer to tries to claim). Sometimes a mixin is better, sometimes a HOC.

Mixins:
- do not introduce unnecessary overhead (instances, rendering)
- directly add (public) behavior to the class; the returned class is a subclass of the original (drop in repl.)
- require more care esp. when attaching behavior to standard callback methods and name pollution
- less effort is needed to expose "original" behavior (props, etc.) 

HOCs:
- are good at hidding and encapsulating behavior
- the returned class is not a drop in replacement of the original. much more work is required to "simulate" that behavior
- add overhead in instances and rendering

In  this particular situation, a mixin is the more appropriate choice.

Reply:
Thank you for the detailed explanation. I also agree that using a mixin is more appropriate.